### ISSUE TITLE
🐛 Bug Report: scaffolder/next import from `/alpha` finds different esm than stable in 1.12

### ISSUE THREAD

**Issue by Pike** [backstage/backstage#17073]
### 📜 Description

When trying to use both the scaffolder stable and scaffolder next, the app frontend fails on finding two different implementations of the scaffolder plugin.

This is due to the `/alpha` is importing from `node_modules/@backstage/*/dist/esm/alpha-714dad1b.esm.js`.

This might be from https://github.com/backstage/backstage/pull/15816, CC @Rugvip 

### 👍 Expected behavior

We should be getting a single implementation for shared code used in alpha and stable.

### 👎 Actual Behavior with Screenshots

The app fails on startup due to conflicting implementations of the scaffolder plugin.

### 👟 Reproduction steps

This affects instances of backstage using both stable and next UI from scaffolder.

It doesn't affect the demo app as the alpha and non-alpha split is only happening on bundled packages, I think

### 📃 Provide the context for the Bug.

Trying to update from 1.11 to 1.12.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by Rugvip** [comment]
Thank you for reporting! #17087 should address this

**Comment by Pike** [comment]
Thanks for the fix. @Rugvip , would this be good for a dot release?

**Comment by Rugvip** [comment]
@Pike could you try the latest `next` release of the scaffolder? I feel a dot release risks having higher impact than leaving this for next mainline.

**Comment by Pike** [comment]
I tried the next.1 release, and it fixes this problem (yay).

The downside is that it's also including the scaffolder permissions (#15798), which I haven't been able to wrap my head around yet.
Also, the zod updates seem to have taken some changes, which I haven't dug into yet. Seems to affect both permissions and scaffolder according to just updating and running `tsc`. Maybe this just requires adjusting plugin dependencies on the same version, but as I mentioned, I didn't dig into this.

Long story short, I understand that changing packaging in a dot release is hefty. Personally, I'd prefer to do the updates above with release notes.
